src/osd/PrimaryLogPG.cc

```diff
@@ -10492,6 +10496,9 @@ int PrimaryLogPG::start_dedup(OpRequestRef op, ObjectContextRef obc)
   for (auto p : chunks) {
     hobject_t target = mop->new_manifest.chunk_map[p.first].oid;
     if (refs.find(target) == refs.end()) {
+      if (target.get_hash() == 0) {
```

Comment:
This check is incorrect, 1 in 1<<32 objects will have a hash of 0.

Reply:
You're right. Thank you for explanation.
It was a depreciated check which was intended to handle former bug fix before you commented.
So, I should have removed this code in advance. 
Sorry for confusing you.